### PR TITLE
Integrate with CCDB5-API - very basic skeleton query

### PR DESCRIPTION
**Pull request by amymok**
This is to first integration with the [CCDB5-API](https://github.com/cfpb/ccdb5-api/) so we have access to the complaint data.

## Additions

- Added index/doc_type environment variable template 
- Added basic calls from CCDB5-API
- Added repository to requirements

## Removals

-

## Changes

-

## Testing

1. This will work after https://github.com/cfpb/ccdb5-api/pull/2 is merged
1. This assumes that you have accessed to the complaint data at this point through elasticsearch
1. Pip install `requirements/optional-public.txt` or just pip install ccdb5-api
1. Update the `COMPLAINT_ES_INDEX` and `COMPLAINT_DOC_TYPE` to the correct index and doctype

## Notes

-

## Todos

- 

## Reviews
@JeffreyMFarley @rosskarchner @willbarton @adamzarger 
cc @sephcoster 

## Checklist

* [ ] PR has an informative and human-readable title
* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [ ] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

**Comment by willbarton** [comment]
I think we misunderstood each other in our discussions yesterday and in the past.

What we've been suggesting is making the ccdb5 Django app separate from cfgov-refresh and including it in [requirements/optional-public.txt](https://github.com/cfpb/cfgov-refresh/blob/master/requirements/optional-public.txt) so that we can isolate the ccdb5-ui dependencies. Do you all still want to do that before we go too far in making modifications here? I think it makes a great deal of sense in this particular case (and the logical separation between CMS-stuff and non-CMS-stuff).

Either way you all decide to go, requirements/optional-public.txt is optional, and doesn't get included in our Travis testing or installed by our setup scripts for local development. If the `ccdb5` Django app is going to live in this repo, `complaint_search` will have to be in [requirements/base.txt](https://github.com/cfpb/cfgov-refresh/blob/master/requirements/base.txt). 

But I really highly recommend having it in its own repository, include it in requirements/optional-public.txt, and have it handle the front-end build using [cfgov_setup](https://github.com/cfpb/cfgov-django-setup) [like the existing `complaint` app](https://github.com/cfpb/complaint/blob/master/setup.py#L37-L38), [regulations-site](https://github.com/cfpb/regulations-site/blob/master/setup.py#L10-L11), etc.

File: cfgov/ccdb5/views.py [hunk 1]
```diff
@@ -1,20 +1,24 @@
 from django.http import JsonResponse
 import datetime
+import complaint_search
 
 
 def search(request):
     now = datetime.datetime.now()
     results = {'search': {'timestamp': '%s' % now}}
```
Comment (JeffreyMFarley) [inline cfgov/ccdb5/views.py]: I think you can get rid of the timestamp lines

**Comment by JeffreyMFarley** [comment]
You may also want to check why the tests failed

File: cfgov/ccdb5/views.py [hunk 2]
```diff
@@ -1,20 +1,24 @@
 from django.http import JsonResponse
 import datetime
+import complaint_search
 
 
 def search(request):
```
Comment (rosskarchner) [inline cfgov/ccdb5/views.py]: I don't understand why these views aren't in ccdb5-api?

File: requirements/optional-public.txt [hunk 3]
```diff
@@ -5,3 +5,4 @@ git+https://github.com/cfpb/owning-a-home-api.git@0.9.96#egg=owning-a-home-api
 git+https://github.com/cfpb/regulations-core.git@1.2.3#egg=regcore
 git+https://github.com/cfpb/regulations-site.git@2.1.8#egg=regulations
 git+https://github.com/cfpb/retirement.git@0.5.5#egg=retirement
+-e git+https://github.com/cfpb/ccdb5-api.git@master#egg=ccdb5-api
```
Comment (rosskarchner) [inline requirements/optional-public.txt]: Please pin a version!

**Comment by rosskarchner** [comment]
One thing you'll want to look at, is the OPTIONAL_APPS dictionary: This makes sure that apps that aren't installed aren't added to INSTALLED_APPS

https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/cfgov/settings/base.py#L84

Similarly, you'll want to use include_if_app_enabled to make sure we don't add URL's for apps that aren't installed: https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/cfgov/urls.py#L241

**Comment by amymok** [comment]
Thanks for all the feedback, things have changed quite a bit based on your comment, I will make some updates to reflect the new reality.

**Comment by amymok** [comment]
@willbarton @rosskarchner I changed the code and tried to put in the `complaint_search` app, but I am still getting a 404 when I navigate to data-research/consumer-complaints/api/v1, is there anything else I am doing incorrectly here or anything missing here?  Thanks~

**Comment by amymok** [comment]
Okay the issue has been fixed, the URLs are now all working.  Thank you @rosskarchner.  I am going to clean up this PR now (look at test failure + merge conflicts)

**Comment by amymok** [comment]
@rosskarchner @willbarton This should be ready for merging

**Pull request merged**